### ISSUE TITLE
Token sampling: temperature, top-p, top-k

### ISSUE THREAD

**Issue by Davenporten** [Davenporten/noddle#5]
## Problem

Token selection is currently pure greedy (argmax over logits). This produces repetitive, low-diversity output — the model always picks the single highest-probability token at every step. Without any sampling, outputs feel mechanical and the model gets stuck in repetition loops.

## What to add

- **Temperature** — divide logits by T before softmax. T < 1 sharpens the distribution (more deterministic), T > 1 flattens it (more random). T=1.0 is neutral.
- **Top-k** — only sample from the k highest-probability tokens. Prevents the model from choosing very unlikely tokens.
- **Top-p (nucleus sampling)** — only sample from the smallest set of tokens whose cumulative probability exceeds p. More adaptive than top-k.
- **Repetition penalty** — reduce the probability of tokens that have already appeared in the context. Addresses the most obvious failure mode of greedy decoding.

## Recommended defaults

```
temperature = 0.7
top_p = 0.9
top_k = 50
repetition_penalty = 1.1
```

## Where it goes

Sampling happens at the final node in the chain, after the LM head produces logits. It replaces the current `argmax_from_wire()` call in `noddle-node/src/state.rs`.

The sampling parameters should be per-request fields in `PromptRequest` with server-side defaults.

## What changes

- `proto/noddle.proto` — add sampling params to `PromptRequest`
- `noddle-adapter-candle/src/tensor_io.rs` — replace `argmax_from_wire` with `sample_from_wire(logits, params)`
- `noddle-node/src/state.rs` — pass sampling params through
- `noddle-cli/src/main.rs` — expose `--temperature`, `--top-p` flags
